packages/astro-angular/src/server.ts

```diff
@@ -1,35 +1,98 @@
 import 'zone.js/dist/zone.js';
-import { renderApplication } from '@angular/platform-server';
-import type { ɵComponentType } from '@angular/core';
+import type {
+  ComponentMirror,
+  Provider,
+  ɵComponentType as ComponentType,
+} from '@angular/core';
+import {
+  ApplicationRef,
+  InjectionToken,
+  reflectComponentType,
+} from '@angular/core';
+import {
+  BEFORE_APP_SERIALIZED,
+  renderApplication,
+} from '@angular/platform-server';
+
+const ANALOG_ASTRO_STATIC_PROPS = new InjectionToken<{
+  props: Record<string, unknown>;
+  mirror: ComponentMirror<unknown>;
+}>('Static props token w/ mirror', {
+  factory() {
+    return { props: {}, mirror: {} as ComponentMirror<unknown> };
+  },
+});
 
 function check(
-  Component: ɵComponentType<unknown>,
+  Component: ComponentType<unknown>,
   _props: Record<string, unknown>,
```

Comment:
```suggestion
   props: Record<string, unknown>,
```

Reply:
We're not using `_props` in `check()` yet so keeping `_props` as is